state-chain/custom-rpc/src/lib.rs

```diff
@@ -124,6 +124,12 @@ pub enum EncodingType {
 	Sol(SolEncodingType),
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Decode, Serialize, Deserialize)]
```

Comment:
would make the API slightly nicer
```suggestion
#[derive(Clone, Debug, PartialEq, Eq, Decode, Serialize, Deserialize)]
#[serde(untagged)]
```